src/runtime/react/render.js

```diff
@@ -1,14 +1,54 @@
 import { runEffects } from "../effects";
 import renderInternal from "../renderInternal";
 
+const flatten = (array) => {
+	const flatArray = [];
+	for(let i = 0; i < array.length; i++) {
+		if (Array.isArray(array[i])) {
+			const flattenedChild = flatten(array[i]);
+			flatArray.splice(flatArray.length, 0, ...flattenedChild);
+		} else {
+			flatArray.push(array[i]);
+		}
+	}
+	return flatArray;
+};
+
+const replaceArray = (nodes, parentNode) => {
+	let shouldReplace = nodes.length !== parentNode.childNodes.length;
+	if (!shouldReplace) {
+		for(let i = 0; i < nodes.length; i++) {
+			if (nodes[i] !== parentNode.childNodes[i]) {
+				shouldReplace = true;
+				break;
+			}
+		}
+	}
+	if (shouldReplace) {
+		for(let i = 0; i < nodes.length; i++) {
+			parentNode.appendChild(nodes[i]);
+		}
+		for(let i = nodes.length; i < parentNode.childNodes.length; i++) {
+			parentNode.removeChild(parentNode.childNodes[0]);
+		}
+	}
+}
+
 const map = new WeakMap();
 export default (element, parentNode) => {
 	let context = map.get(parentNode);
 	if (!context) {
 		map.set(parentNode, (context = { $$: { node: parentNode } }));
 	}
 	renderInternal(context, element, "node", false);
-	const node = context.node;
-	if (node.parentElement !== parentNode) parentNode.appendChild(node);
+	let node = context.node;
```

Comment:
This duplicates a lot of logic from `src/runtime/replaceNode`. You should be able to reuse this logic.

Reply:
aha, yes. I just had to fix replaceNode to recognize NodeList as an array..